services/contacts/js/app.js

```diff
@@ -0,0 +1,165 @@
+(function(window) {
+  'use strict';
+
+  function debug(str) {
+    console.log('ContactsService -*-:' + str);
+  }
+
+  // Ok, this kinda sucks because most APIs (and settings is one of them) cannot
+  // be accessed from outside the main thread. So basically everything has to go
+  // down to the SW thread, then back up here for processing, then back down to
+  // be sent to the client. Yay us!
+
+  var processSWRequest = function(channel, evt) {
+
+    var _contacts = navigator.mozContacts;
+    // We can get:
+    // * oncontactchange
+    // * getAll
+    // * methodName
+    var remotePortId = evt.data.remotePortId;
+    var request = evt.data.remoteData;
+    var requestOp = request.data;
+
+    function _cloneContact(mozContact) {
```

Comment:
I don't understand your explanation, Manu, sorry. Do you mean we can't clone Date objects? Wouldn't it be better to add this check to the standard clone object so it doesn't bite us on the ass later?